### DISCUSSION TITLE
Enable full error tracing in production builds

### DISCUSSION THREAD

**Discussion by CaptainN** [vercel/next.js#66146] [App Router]
How can I enable full error read outs in production builds? I've run in to half a dozen production errors in the last few weeks on the latest versions of nextjs that cannot be reproduced in dev mode - very aggravating - but the errors in production logs are obfuscated, which is frustrating (even in server logs - which is beyond frustrating).

My builds are in staging environments with various kinds of protection, so there's no security issue. How do I enable regular errors?

this should be as easy as putting something in `next.config.js` (or maybe a compile flag similar to  `--no-mangling`) - does such an option exist?

This error (in SERVER LOGS in a local build running on my machine) is not so useful:
```
 ⨯ [Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.] {
  digest: 'DYNAMIC_SERVER_USAGE'
}
```

**Comment by juligris** [comment]
I am looking for the exact same answer. I need the non-minified errors in my production build to troubleshoot. I am not worried about security in my instance as well.

**Comment by oleksandr-tkalenko** [comment]
Absolutely agree. I recently ran into the similar case. There should be an option to enable descriptive error details for debugging purposes in production builds.

**Comment by izucken** [comment]
Maybe this will help you as id did me. I ran into this and was baffled that its not a default, previously working in different languages I **always** had useful stack traces on the backend side of things, or made sure that I had them which was usually easy and obvious how to do.

By default server side stack trace is useless in comparison, e.g.:
```
TypeError: e.filter is not a function
  at i (/workspace/.next/server/app/.../page.js:1:17179)
  at ee (/workspace/.next/server/app/.../page.js:1:13418)
```

To enable better tracing for next I did this after copious amounts of googling:
1. Ensure nodejs is run with `--enable-source-maps` flag. Since in my case I also need a custom server, my run command looks like this: `NODE_OPTIONS='--enable-source-maps' NODE_ENV=production node src/server.js`
2. Ensure `serverSourceMaps` option is on in `next.config.mjs` config:
```
const nextConfig = {
    experimental: {
        serverSourceMaps: true,
    },
};
```

Take note that both node flag and nextjs option are marked as experimental, so there is no good guarantees of this working properly.

After this config my stack traces becomes more useful:
```
TypeError: e.filter is not a function
  at myFunctionName (webpack://project/src/.../realSource.ts:74:31)
  at ee (webpack://project/src/app/.../page.tsx:156:21)
```

**Reply by khanakia** [reply]
Didn't work
